File: code2040_step3_aburt.py
# CODE2040 Step Three Assessment Challenge
import requests
import json

# CODE2040 API Token.
myAPIToken = '3624f3214a21f60da7488a0716e6eee3'

# CODE2040 Step 3 Challenge URL.
codeChallengeUrl = 'http://challenge.code2040.org/api/haystack'

# Validation URL for Step 3 challenge.
codeValidationUrl = 'http://challenge.code2040.org/api/haystack/validate'

# POST Request for challenge dictionary and Response.
step3 = requests.post(codeChallengeUrl, data={'token': myAPIToken})
print(step3.text)

# Converts POST Request to JSON.
step3jsonResponse = json.loads(step3.text)

# Isolation of target String from API dictionary.
queryNeedle = step3jsonResponse["needle"]
print(queryNeedle)

# Returns index of the target String found in List.
foundNeedle = step3jsonResponse["haystack"].index(queryNeedle)
print(foundNeedle)

# Dictionary to POST for Validation.
dataToPost = {
    'token': myAPIToken,
    'needle': foundNeedle
}
# API Challenge Validation and Response.
step3Validate = requests.post(codeValidationUrl, json=dataToPost)
print(step3Validate.text)

File: code2040_step3_aburt.js

var Promise = require("bluebird");
var http = require("http");
http.post = require("http-post");
var url = "http://challenge.code2040.org/api/haystack";
var myAPIToken = '3624f3214a21f60da7488a0716e6eee3';
var validateUrl = "http://challenge.code2040.org/api/haystack/validate";
var infoToken= {
	'token': myAPIToken
}

function result(){ 
	return new Promise(function (resolve){
		var requestResponse = http.post(url,infoToken, function(response){
			response.on('data', function(data) {
				resolve (data);
			}).setEncoding('utf8');
			response.on("error",function(error){
				console.log(error);
			})
		});
	})
};

result().then(function(data){
	console.log(data);
	data = JSON.parse(data);
	var needle = data["needle"];
	var haystack = data["haystack"];
	console.log(needle);
	console.log(haystack);
	var foundNeedle = haystack.indexOf(needle);
	console.log(foundNeedle);


	var dataToPost = {
		'token': myAPIToken,
		'needle': foundNeedle

	};
	http.post(validateUrl,dataToPost,function(response){
		response.on("data", function(data){
			console.log(data);
		}).setEncoding("utf8");
	});

});
